### ISSUE TITLE
Expose delimiters for `uri_autolink`

### ISSUE THREAD

**Issue by musjj** [MDeiml/tree-sitter-markdown#109]
I want to make a query to conceal the delimiters for `uri_autolink` (the angle brackets in `<https://www.google.com>`), but the delimiters are not exposed in the tree.

For example for emphasis markers, you have this tree:

```scm
emphasis [12, 2] - [12, 8]
  emphasis_delimiter [12, 2] - [12, 3]
  emphasis_delimiter [12, 7] - [12, 8]
```

Can we have something similar for `uri_autolink`?

**Comment by clason** [comment]
If you're using `:InspectTree`, you need to toggle anonymous nodes:
```
(section) ; [1:1 - 2:0]
 (paragraph) ; [1:1 - 2:0]
  (inline) ; [1:1 - 24]
   (uri_autolink) ; [1:1 - 24]
   "<" ; [1:1 - 1]
   ":" ; [1:7 - 7]
   "/" ; [1:8 - 8]
   "/" ; [1:9 - 9]
   "." ; [1:13 - 13]
   "." ; [1:20 - 20]
   ">" ; [1:24 - 24]
```
so you can just match them literally.

**Comment by musjj** [comment]
Hmm, is there a way to match literal characters in treesitter queries? I tried:

```lisp
(uri_autolink
  [
    "<"
    ">"
  ] @conceal
  (#set! conceal ""))
```

But it's giving me errors.

**Comment by clason** [comment]
they're not children of `uri_autolink`, so you'd have to write a more complicated query using anchors

**Comment by musjj** [comment]
I tried to use `match?`, but it doesn't really help because it will still apply `@conceal` to the whole node. Is there really a way to apply it partially? I also tried `offset!`, but I don't think you can capture the individual delimiters this way.

**Comment by clason** [comment]
Yes, you need to capture the anonymous nodes themselves. You might need two individual queries, one for `<` and one for `>`. (`#offset!` could help with that).

Of course, you could also make a PR adding `uri_autolink_delimiter` nodes to the grammar.

**Comment by MDeiml** [comment]
I think implementing a `uri_autolink_delimiter` would be quite complex, so I recommend sticking with the `#offset!` plan. If I see correctly though offset doesn't quite do what you want so the best way to go is probably to [write your own directive](https://neovim.io/doc/user/treesitter.html#vim.treesitter.query.add_directive()).

**Comment by clason** [comment]
There's also `#gsub!` for more complex transformations. (Not cheap, though!)
